rotkehlchen/exchanges/bitfinex.py

```diff
@@ -0,0 +1,933 @@
+                )
+                return []
+
+            raise AssertionError(f'Unexpected {self.name} response_case: {case}') from e
+
+        error_data = self._get_error_response_data(response_list)
+        self._check_for_system_clock_not_synced_error(error_data.error_code)
+        # Errors related with the API key return a human readable message
+        if case == 'validate_api_key' and error_data.error_code == API_KEY_ERROR_CODE:
+            return False, API_KEY_ERROR_MESSAGE
+
+        # Below any other error not related with the system clock or the API key
+        reason = error_data.reason or response.text
+        msg = (
+            f'{self.name} query responded with error status code: {response.status_code} '
+            f'and text: {reason}.'
+        )
+        log.error(msg)
+
+        if case == 'validate_api_key':
+            raise RemoteError(msg)
+        if case == 'balances':
+            return None, msg
+        if case in {'trades', 'asset_movements'}:
+            self.msg_aggregator.add_error(
+                f'Got remote error while querying {self.name} {case}: {msg}',
+            )
+            return []
+
+        raise AssertionError(f'Unexpected {self.name} response_case: {case}')
+
+    def first_connection(self) -> None:
+        self.first_connection_made = True
+
+    @protect_with_lock()
+    @cache_response_timewise()
+    def query_balances(self) -> Tuple[Optional[Dict[Asset, Balance]], str]:
+        """Return the account exchange balances on Bitfinex
+
+        The wallets endpoint returns a list where each item is a currency wallet.
+        Each currency wallet has type (i.e. exchange, margin, funding), currency,
+        balance, etc. Currencies (tickers) are in Bitfinex format and must be
+        standardized.
+
+        Endpoint documentation:
+        https://docs.bitfinex.com/reference#rest-auth-wallets
+        """
+        try:
+            currency_map_response = self._query_currency_map()
+        except IndexError:
+            msg = f'Error processing {self.name} currency map. Response is empty'
+            log.error(msg)
+            return None, msg
+
+        if currency_map_response.success is False:
+            result, msg = self._process_unsuccessful_response(
+                response=currency_map_response.response,
+                case='balances',
+            )
+            return result, msg
+
+        response = self._api_query('wallets')
+        if response.status_code != HTTPStatus.OK:
+            result, msg = self._process_unsuccessful_response(
+                response=response,
+                case='balances',
+            )
+            return result, msg
+        try:
+            response_list = rlk_jsonloads_list(response.text)
+        except JSONDecodeError as e:
+            msg = f'{self.name} returned invalid JSON response: {response.text}.'
+            log.error(msg)
+            raise RemoteError(msg) from e
+
+        # Wallet items indexes
```

Comment:
nitpick: indexes -> indices

Reply:
wow thanks! I didn't know. It is like in Spanish